src/Logger/Logger.cc

```diff
@@ -14,7 +14,11 @@ namespace fs = boost::filesystem;
 namespace fs = std::filesystem;
 #endif
 
-void InitLogger(bool no_log_file, int verbosity) {
+static bool log_protocol_messages(false);
+
+void InitLogger(bool no_log_file, int verbosity, bool log_performance, bool this_log_protocol_messages) {
+    log_protocol_messages = this_log_protocol_messages;
```

Comment:
I meant that you could use the `this` pointer to refer to the member variable, and call the parameter the same thing:

    this->log_protocol_messages = log_protocol_messages

Reply:
But this syntax would result in a compile error. Or just create a Logger class?

Reply:
Sorry; I didn't notice that the function is not a class member. The solution you had before (with the underscore at the end) was fine!